cluster/src/main/scala/io/snappydata/impl/LeadImpl.scala

```diff
@@ -239,6 +243,25 @@ class LeadImpl extends ServerImpl with Lead
     }
   }
 
+  private def checkAuthProvider(props: Properties): Unit = {
+    var authP = props.getProperty(Attribute.AUTH_PROVIDER)
+    if (authP == null) {
+      authP = props.getProperty(Attribute.SERVER_AUTH_PROVIDER)
+    }
+    if (authP != null && !"LDAP".equalsIgnoreCase(authP)) {
+      throw new IllegalArgumentException("LDAP is the only supported auth-provider currently.")
+    } else if (authP != null && !isEnterpriseEdition()) {
+      throw new UnsupportedOperationException("Security feature is available in SnappyData " +
```

Comment:
Hmm, thinking was that anything other than LDAP is not legal currently, while auth-provider itself is unsupported for OSS. No strict requirement, both could be throwing same type.